spec/acceptance/splunk_spec.rb

```diff
@@ -5,6 +5,11 @@
     # Using puppet_apply as a helper
     it 'works idempotently with no errors' do
       pp = <<-EOS
+      class { '::splunk::params':
+        version => '7.0.0',
+        build   => 'c8a78efdd40f',
+        src_root => 'https://download.splunk.com',
```

Comment:
This should be unnecessary, no? (since it's the same as params)?

More generally, I'm not sure about passing stuff into params directly, though I know the module did this before as well.

Reply:
I can remove this since i did change it to the defaults.  Shall i remove this?

Also yes this module uses params in a non-standard way.  Partly in fact because both the forwarder and the server try and reuse the params class.

Reply:
I saw that postgresql module uses `postgresql::global` or something along those lines to work around setting things that are common to multiple classes (and then pulls those in / out of params as necessary) - not sure if that's a more standard pattern though... most modules these days seem to be leaning towards having the main class be the interface to everything, and then having toggles there (`enable_forwarder = true` in class `splunk`). Probably not worth changing in this PR, but might be worth asking about, and possibly changing down the road.

Reply:
Yep I agree there are a couple different approaches.  That is a beast for another day!